### ISSUE TITLE
production lists apply smart quotes transform since Sphinx 1.6.1

### ISSUE THREAD

**Issue by jfbu** [sphinx-doc/sphinx#3824]
With following source:
```
.. productionlist::
   try_stmt: try1_stmt | try2_stmt
   try1_stmt: "try" ":" `suite`
            : ("except" [`expression` ["," `target`]] ":" `suite`)+
            : ["else" ":" `suite`]
            : ["finally" ":" `suite`]
   try2_stmt: "try" ":" `suite`
            : "finally" ":" `suite`
```

and Sphinx 1.6.1 or 1.6.2 the pseudoxml produces:

```
        <productionlist>
            <production ids="grammar-token-try_stmt" tokenname="try_stmt">
                 try1_stmt | try2_stmt
            <production ids="grammar-token-try1_stmt" tokenname="try1_stmt">
                 “try” “:” 
                <literal classes="xref">
                    suite
            <production tokenname="">
                 (“except” [
                <literal classes="xref">
                    expression
                 [“,” 
                <literal classes="xref">
```

and the curly quotes are observed in html output too.

### Analysis

(corrected)

The 1.5.6 ``SmartyPantsHTMLTranslator`` set a flag  ``no_smarty`` on visiting ``desc_signature`` or ``productionlist`` nodes, which was then inherited by all descendants.

When proceeding via Docutils, the ``TextElement`` block-level nodes are traversed. ``productionlist`` isn't such a node, but ``production`` is. And ``production`` node was still missing from the #3816 SphinxSmartQuotes ``get_tokens`` method extra list of nodes to skip. 

The issue applies also potentially with ``option`` directive or general object signatures regarding ``desc_addname`` node as it was not explicitely added at #3816.

Relates: #3666, #3816, #3818, https://github.com/sphinx-doc/sphinx/pull/3816#discussion_r118893735

Followed-up in #3826

### environment
- Sphinx 1.6.1 or 1.6.2

**Comment by jfbu** [comment]
If you encounter the issue and need a temporary workaround, add a file ``docutils.conf`` to top repertory with following contents:

```
[parsers]
smart_quotes: no
```

This will globally deactivate smart quotes. The deprecated ``conf.py`` config setting ``html_use_smartypants = False`` can also be used, but this will emit a deprecation warning.

**Comment by jfbu** [comment]
Fixed at f99fe20e. Refer to #3826 for further discussion.
